app/models/notification.rb

```diff
@@ -120,20 +120,20 @@ def send_milestone_notification_without_delay(type:, article_id:)
       Notifications::MilestoneJob.perform_now(type, article_id)
     end
 
-    def remove_all(notifiable_id:, notifiable_type:, action: nil)
-      Notifications::RemoveAllJob.perform_later(notifiable_id, notifiable_type, action)
+    def remove_all_by_action(notifiable:, action: nil)
+      Notifications::RemoveAllByActionJob.perform_later(notifiable, action)
     end
 
-    def remove_all_without_delay(notifiable_id:, notifiable_type:, action: nil)
-      Notifications::RemoveAllJob.perform_now(notifiable_id, notifiable_type, action)
+    def remove_all_by_action_without_delay(notifiable:, action: nil)
+      Notifications::RemoveAllByActionJob.perform_now(notifiable, action)
     end
 
-    def remove_each(notifiable_collection)
-      Notifications::RemoveEachJob.perform_later(notifiable_collection.pluck(:id))
+    def remove_all(notifiable_collection)
+      Notifications::RemoveAllJob.perform_later(notifiable_collection)
```

Comment:
Same as above. the job itself declares that it's asking for IDs but we're passing the whole collection here, it should be only ids, and then they get loaded by the job itself.

I think this should should receive the type of collection, eg. `Comment` and the list of ids to load from the DB

You can see an example here https://github.com/thepracticaldev/dev.to/blob/710c1d9e7b24a3e0b144bec6b3f22d2835cf4c4e/app/jobs/follows/touch_follower_job.rb#L5